From bc1c2bad255e33d12831c66be5770692b2519972 Mon Sep 17 00:00:00 2001
From: Sunny Titus <sunny.titus@deltares.nl>
Date: Wed, 18 Dec 2024 11:10:35 +0100
Subject: [PATCH] Add multidoc support to older versions

---
 docs/conf.py | 24 +++++++++++++++++++++++-
 1 file changed, 23 insertions(+), 1 deletion(-)

diff --git a/docs/conf.py b/docs/conf.py
index cc6136e8..98e9ce44 100644
--- a/docs/conf.py
+++ b/docs/conf.py
@@ -8,7 +8,8 @@
 
 # -- Path setup --------------------------------------------------------------
 
-
+import os
+import subprocess
 from importlib.metadata import distribution
 
 # If extensions (or modules to document with autodoc) are in another directory,
@@ -126,6 +127,20 @@ html_context = {"default_mode": "light"}
 # further.  For a list of options available for each theme, see the
 # documentation.
 #
+version_or_name = subprocess.run(
+    "git symbolic-ref -q --short HEAD || git describe --tags --exact-match",
+    capture_output=True,
+    shell=True,
+    text=True,
+).stdout.strip()
+
+env = os.environ
+json_url = (
+    "https://deltares.github.io/imod-python/_static/switcher.json"
+    if "JSON_URL" not in env
+    else env["JSON_URL"]
+)
+
 html_theme_options = {
     "navbar_align": "content",
     "icon_links": [
@@ -146,6 +161,13 @@ html_theme_options = {
         "image_light": "imod-python-logo-light.svg",
         "image_dark": "imod-python-logo-dark.svg",
     },
+    "switcher": {
+        "json_url": json_url,
+        "version_match": version_or_name,
+    },
+    "navbar_end": ["theme-switcher", "navbar-icon-links", "version-switcher"],
+    "show_version_warning_banner": True,
+    "check_switcher": False,
 }
 
 # Custom sidebar templates, must be a dictionary that maps document names
-- 
2.47.1.windows.1

